src/hardware/serialport/misc_util.cpp

```diff
@@ -31,6 +31,15 @@
 // Constants
 constexpr int connection_timeout_ms = 5000;
 
+std::string to_string(const SocketType socket_type)
+{
+	switch (socket_type) {
+	case SocketType::Tcp: return "TCP";
+	case SocketType::Enet: return "ENet";
+	default: assert(false); return "(unknown)";
+	}
+}
```

Comment:
suggest returning a `const char* const`, because that's what they are (all `"string_literal"s` are permanent and pinned into the executable, their memory and accessibility is permanent)

Reply:
Will also cleanup the need for re-converting back to the `char*`s in a couple places:

to_string(socketType).c_str());